src/decomp_2d_nccl.f90

```diff
@@ -28,8 +28,13 @@ module decomp_2d_nccl
    type(ncclComm), save, public :: nccl_comm_2decomp
    integer(kind=cuda_stream_kind), save, public :: cuda_stream_2decomp
 
+   ! Extra pointers for nccl complex transpose
+   real(mytype), dimension(:), device, public, pointer, contiguous :: work3_r_d, work4_r_d
```

Comment:
If we move `work3` and `work4` here, we could have `allocatable` instead of `pointer, contiguous`.

I think the public attribute is not needed if we move `work3` and `work4` here

Reply:
yes we can do that. I just wanted to be consistent with the rest.

Reply:
If possible, I think it is best to define variables in the module where they are needed.

Reply:
I think it's best to move them here:
1) They're specific to the NCCL backend so it's neater
2) They're intended to be temporary until we can fix the NCCL issue - prevents people from using them for something else (which will then be hard to untangle).